torchmultimodal/modules/layers/quantisation.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import nn, Tensor
+
+
+class Quantisation(nn.Module):
+    """Quantisation provides an embedding layer that takes in a collection of flattened vectors, usually the
+    output of an encoder architecture, and performs a nearest-neighbor lookup in the embedding space.
+
+    Vector quantisation was introduced in Oord et al. 2017 (https://arxiv.org/pdf/1711.00937.pdf) to generate high-fidelity
+    images, videos, and audio data.
+
+    Args:
+        num_embeddings (int): the number of vectors in the embedding space
+        embedding_dim (int): the dimensionality of the embedding vectors
+    """
+
+    def __init__(self, num_embeddings: int, embedding_dim: int):
+        super().__init__()
+        self.embedding_dim = embedding_dim
+        self.num_embeddings = num_embeddings
+
+        self.embedding = nn.Embedding(self.num_embeddings, self.embedding_dim)
+        self.embedding.weight.data.uniform_(
```

Comment:
The original [paper](https://papers.nips.cc/paper/2017/file/7a98af17e63a0ac09ce2e96d03992fbc-Paper.pdf) assumes uniform distribution to be the prior for the latent vectors `p(z)` such that there's equal probability for each latent vector `z`. The posterior `p(z|x)` is deterministic by way of minimizing the L2 distance.

Constraining the values within `z` to be uniform is not prescribed per se but (I guess) it probably makes the convergence with the encoder output easier.

cc @ebsmothers

Reply:
I've seen this in at least three separate implementations of the model. It probably makes more sense than initializing with normal distribution anyway.